### PR TITLE
feat: Add `state` to `aws_cloudwatch_event_rule`

### PR DESCRIPTION
**Pull request by magreenbaum**
## Description
Adds `state` to `aws_cloudwatch_event_rule` and deprecates `is_enabled`. 
 Changed the default of `is_enabled` to `null` since it is default `true` by the provider already and allows for the use of `state`. 

## Motivation and Context
https://github.com/hashicorp/terraform-provider-aws/pull/34510
Fixes https://github.com/terraform-aws-modules/terraform-aws-eventbridge/issues/103

## Breaking Changes
No. Left `is_enabled` in place so it's backwards compatible.

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: examples/complete/main.tf [hunk 1]
```diff
@@ -44,17 +44,25 @@ module "eventbridge" {
     orders = {
       description   = "Capture all order data"
       event_pattern = jsonencode({ "source" : ["myapp.orders"] })
-      enabled       = false
+      state         = "DISABLED" # conflicts with is_enabled which is deprecated
```
Comment (magreenbaum) [inline examples/complete/main.tf:47]:
```suggestion
      state         = "DISABLED" # conflicts with enabled which is deprecated
```

File: examples/complete/main.tf [hunk 2]
```diff
@@ -44,17 +44,25 @@ module "eventbridge" {
...
     }
     emails = {
       description   = "Capture all emails data"
       event_pattern = jsonencode({ "source" : ["myapp.emails"] })
-      enabled       = true
+      state         = "ENABLED" # conflicts with is_enabled which is deprecated
```
Comment (magreenbaum) [inline examples/complete/main.tf:52]:
```suggestion
      state         = "ENABLED" # conflicts with enabled which is deprecated
```

File: examples/with-api-destination/main.tf [hunk 3]
```diff
@@ -23,7 +22,7 @@ module "eventbridge" {
     orders = {
       description   = "Capture all order data"
       event_pattern = jsonencode({ "source" : ["myapp.orders"] })
-      enabled       = true
+      state         = "ENABLED" # conflicts with is_enabled which is deprecated
```
Comment (magreenbaum) [inline examples/with-api-destination/main.tf:25]:
```suggestion
      state         = "ENABLED" # conflicts with enabled which is deprecated
```

File: examples/with-ecs-scheduling/main.tf [hunk 4]
```diff
@@ -45,7 +44,7 @@ module "eventbridge" {
   rules = {
     orders = {
       description         = "Cron for Orders"
-      enabled             = false
+      state               = "DISABLED" # conflicts with is_enabled which is deprecated
```
Comment (magreenbaum) [inline examples/with-ecs-scheduling/main.tf:47]:
```suggestion
      state               = "DISABLED" # conflicts with enabled which is deprecated
```

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 3.1.0 :tada:

**Comment by doctornkz-intelas** [comment]
@antonbabenko thank you for your effort and Slava Ukraini!

**Pull request merged**